### DISCUSSION TITLE
Feature Request: Increase MRU on recent models/files menu and allow us to pin items

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#1337] [Ideas]
### Is your feature request related to a problem? Please describe

The current MRU list is just the most recent 10 models or files opened (queries, pivot grids, diagrams, etc)

I'd like more than 10 (at least on the files, less important on models). When I am working on 2-3 models in the course of a week and saving queries, diagrams, pivot grids, DAX scripts, etc, I can easily burn through 10 file names in a few days. There are some queries I routinely use for model diagnostics (checking incremental refresh partitions against source data for example) that I'd like to always be on that list, even if I haven't used it in a week or two.

### Describe the solution you'd like

A user configurable MRU list up to 50 (arbitrary number, but other apps use that number and 50 is good)

Ability to pin items so they never drop off of the list.

### Goals

1.
2.

**Comment by MariaJoseFF** [comment]
Hi @edhans ,

Thank you for your suggestion. We have added this task in our backlog and we will get back to you once we have an update.

BR Maria Jose
Tabular Editor

**Comment by otykier** [comment]
Hi @edhans 
Happy to let you know that recent models/files can now be pinned/unpinned in v. [3.19.0](https://docs.tabulareditor.com/te3/other/release-notes/3_19_0.html?tabs=downloads). Moreover, you can remove individual items from the list. You can also clear the full lists or change the number of items shown under **Tools > Preferences > User Interface**.
